pkg/operators/ebpf/ebpf.go

```diff
@@ -559,13 +559,27 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 			i.links = append(i.links, l)
 		}
 
-		// we need to store links to iterators on a separated list because we need them to run the programs.
+		// We need to store iterators' because we need them to run the programs
 		if p.Type == ebpf.Tracing && strings.HasPrefix(p.SectionName, iterPrefix) {
 			lIter, ok := l.(*link.Iter)
 			if !ok {
+				i.Close()
 				return fmt.Errorf("link is not an iterator")
 			}
-			i.linksSnapshotters = append(i.linksSnapshotters, &linkSnapshotter{link: lIter, typ: p.AttachTo})
+
+			for _, snapshotter := range i.snapshotters {
+				if _, ok := snapshotter.iterators[progName]; ok {
+					snapshotter.iterators[progName] = &linkSnapshotter{
+						link: lIter,
+						typ:  p.AttachTo,
+					}
+					continue programs
```

Comment:
If we ever need to add more logic after L584, this `continue` will skip it. IMO it's better to use a "found" variable and return error if found is false. 

btw, it is really a hard error to have an iterator that is not associated to any snapshotter? Perhaps we can just print a warning and ignore it.

Reply:
> btw, it is really a hard error to have an iterator that is not associated to any snapshotter? Perhaps we can just print a warning and ignore it.
>
Mmmmm it should be at least a warning because, currently, only snapshotters run iterators so we need to notify users that one of the programs in the eBPF code will not be run.